### DISCUSSION TITLE
Argo Rollout with canary strategy and workloadRef immediately creates a new replicaset

### DISCUSSION THREAD

**Discussion by spatelGDOT** [argoproj/argo-rollouts#3856] [Q&A]
I am currently trying to experiment with argo rollouts.  I am using workloadRef with the canary strategy.  As soon as I apply the rollout it creates a new replicaset in my existing deployment as "stable". The previous replicaSet is still up and never goes down, even when I rollout my first new image.  I would have thought that with workloadRef the replicaset that is already up is the one that would be "stable".  Is this normal?

**Comment by kostis-codefresh** [comment]
Have you scaled down the existing deployment to zero?

Like deployments you can also change the revisionHistoryLimit property if you don't want to see old replicasets around (they will have zero replicas anyway)
